core/templates/services/html-length.service.ts

```diff
@@ -94,7 +103,14 @@ export class HtmlLengthService {
   private calculateBaselineLength(
       sanitizedHtml: string, calculationType: CalculationType): number {
     let domparser = new DOMParser();
-    let dom = domparser.parseFromString(sanitizedHtml, 'text/html');
+    let dom: Document;
+    try {
+      dom = domparser.parseFromString(sanitizedHtml, 'text/html');
+    } catch (error) {
+      throw new Error(
+        'Failed to parse HTML string.' +
+        'Ensure valid HTML tags string is provided.');
```

Comment:
Ensure that a valid string that includes HTML tags is provided.

Reply:
done!